rcl/include/rcl/graph.h

```diff
@@ -581,6 +582,95 @@ rcl_count_subscribers(
   const char * topic_name,
   size_t * count);
 
+/// Wait for there to be a specified number of publishers on a given topic.
+/**
+ * The `node` parameter must point to a valid node.
+ *
+ * The `allocator` parameter must point to a valid allocator.
+ *
+ * The `topic_name` parameter must not be `NULL`, and must not be an empty string.
+ * It should also follow the topic name rules.
+ *
+ * This function blocks and will return when the number of publishers for `topic_name`
+ * is greater than or equal to the `count` parameter, or the specified `timeout` is reached.
+ *
+ * The `timeout` parameter is in nanoseconds.
+ * A negative value disables the timeout (i.e. this function blocks until the number of
+ * publishers is greater than or equals to `count`).
```

Comment:
might want to mention the timeout is based on steady time elapsed, rather than system time.

Reply:
Done: 0b94ee5